File: 3185. Count Pairs That Form a Complete Day II/Solution.py
class Solution:
    def countCompleteDayPairs(self, hours: List[int]) -> int:
        pairs = 0
        seen = [0]*24
        for x in hours:
            x %= 24
            if x:
                pairs += seen[24-x] 
            else:
                pairs += seen[0]
            seen[x] += 1
        return pairs

File: 3185. Count Pairs That Form a Complete Day II/Solution.go
func countCompleteDayPairs(hours []int) int64 {
    seen := [24]int64{}
    var pairs int64 = 0
    for _, val := range hours {
        val %= 24
        if val > 0 {
            pairs += seen[24-val]
        } else {
            pairs += seen[0]
        }
        seen[val] += 1
    }
    return pairs
}
